File: .claude/skills/agent-skill-kit-commons/packages/py/src/skills_ref/lint/neutrality.py
from __future__ import annotations

from pathlib import Path

_PREFIXES_PATH = (
    Path(__file__).resolve().parent.parent / "assets" / "vendor_prefixes.txt"
)

_PREFIXES: list[str] | None = None


def _load_prefixes() -> list[str]:
    global _PREFIXES
    if _PREFIXES is None:
        _PREFIXES = [
            s.strip()
            for s in _PREFIXES_PATH.read_text(encoding="utf-8").split("\n")
            if s.strip()
        ]
    return _PREFIXES


_ALLOWED_TOP_LEVEL = {
    "name",
    "description",
    "tier",
    "license",
    "compatibility",
    "metadata",
    "allowed-tools",
    "dependencies",
}


def check_vendor_neutrality(parsed) -> list[dict]:
    issues: list[dict] = []
    prefixes = _load_prefixes()
    raw = getattr(parsed, "raw_frontmatter", None) or {}
    for key in raw.keys():
        if key in _ALLOWED_TOP_LEVEL:
            continue
        if any(key.startswith(p) for p in prefixes):
            issues.append(
                {
                    "code": "L001_VENDOR_SPECIFIC_FIELD",
                    "severity": "warn",
                    "field": key,
                    "message": (
                        f"'{key}' is vendor-specific; move under 'metadata' "
                        "to preserve portability across agents"
                    ),
                }
            )
    return issues

File: .claude/skills/agent-skill-kit-commons/packages/mjs/src/lint/neutrality.mjs
import { readFileSync } from "node:fs";
import { resolve, dirname } from "node:path";
import { fileURLToPath } from "node:url";

const __dirname = dirname(fileURLToPath(import.meta.url));
const PREFIXES_PATH = resolve(__dirname, "..", "..", "assets", "vendor_prefixes.txt");

let _prefixes = null;
function loadPrefixes() {
  if (_prefixes) return _prefixes;
  _prefixes = readFileSync(PREFIXES_PATH, "utf8")
    .split("\n")
    .map((s) => s.trim())
    .filter(Boolean);
  return _prefixes;
}

const ALLOWED_TOP_LEVEL = new Set([
  "name",
  "description",
  "tier",
  "license",
  "compatibility",
  "metadata",
  "allowed-tools",
  "dependencies",
]);

export function checkVendorNeutrality(parsed) {
  const issues = [];
  const ps = loadPrefixes();
  const fm = parsed?.raw_frontmatter ?? {};
  for (const key of Object.keys(fm)) {
    if (ALLOWED_TOP_LEVEL.has(key)) continue;
    if (ps.some((p) => key.startsWith(p))) {
      issues.push({
        code: "L001_VENDOR_SPECIFIC_FIELD",
        severity: "warn",
        field: key,
        message: `'${key}' is vendor-specific; move under 'metadata' to preserve portability across agents`,
      });
    }
  }
  return issues;
}
